torchaudio/functional.py

```diff
@@ -40,52 +37,44 @@ def scale(tensor, factor):
 
 
 @torch.jit.script
-def pad_trim(tensor, ch_dim, max_len, len_dim, fill_value):
-    # type: (Tensor, int, int, int, float) -> Tensor
-    r"""Pad/trim a 2D tensor (signal or labels).
+def pad_trim(waveform, max_len, fill_value):
+    # type: (Tensor, int, float) -> Tensor
+    r"""Pad/trim a 2D tensor
 
     Args:
-        tensor (torch.Tensor): Tensor of audio of size (n, c) or (c, n)
-        ch_dim (int): Dimension of channel (not size)
-        max_len (int): Length to which the tensor will be padded
-        len_dim (int): Dimension of length (not size)
+        waveform (torch.Tensor): Tensor of audio of size (c, n)
+        max_len (int): Length to which the waveform will be padded
         fill_value (float): Value to fill in
 
     Returns:
         torch.Tensor: Padded/trimmed tensor
     """
-    if max_len > tensor.size(len_dim):
-        # array of [padding_left, padding_right, padding_top, padding_bottom]
-        # so pad similar to append (aka only right/bottom) and do not pad
-        # the length dimension. assumes equal sizes of padding.
-        padding = [max_len - tensor.size(len_dim)
-                   if (i % 2 == 1) and (i // 2 != len_dim)
-                   else 0
-                   for i in [0, 1, 2, 3]]
+    n = waveform.size(1)
+    if max_len > n:
         # TODO add "with torch.no_grad():" back when JIT supports it
-        tensor = torch.nn.functional.pad(tensor, padding, "constant", fill_value)
-    elif max_len < tensor.size(len_dim):
-        tensor = tensor.narrow(len_dim, 0, max_len)
-    return tensor
+        waveform = torch.nn.functional.pad(waveform, (0, max_len - n), 'constant', fill_value)
+    else:
+        waveform = waveform[:, :max_len]
+    return waveform
 
 
 @torch.jit.script
-def downmix_mono(tensor, ch_dim):
-    # type: (Tensor, int) -> Tensor
-    r"""Downmix any stereo signals to mono.
+def downmix_mono(waveform):
+    # type: (Tensor) -> Tensor
+    r"""Downmix stereo waveform to mono. Consider using a `SoxEffectsChain` with
+    the `channels` effect instead of this transformation.
 
     Args:
-        tensor (torch.Tensor): Tensor of audio of size (c, n) or (n, c)
-        ch_dim (int): Dimension of channel (not size)
+        waveform (torch.Tensor): Tensor of audio of size (c, n)
 
     Returns:
-        torch.Tensor: Mono signal
+        torch.Tensor: Tensor that has been downmixed of size (1, n)
     """
-    if not tensor.is_floating_point():
-        tensor = tensor.to(torch.float32)
+    if not waveform.is_floating_point():
```

Comment:
I'm not a big fan of these implicit conversions. What if the users isn't aware this needs floats? Also, it'll return floats for an integer input. Didn't we agree on all waveforms to be floats in range -1 to 1?

Reply:
Done.